### PR TITLE
Fix glacier table integration test

### PR DESCRIPTION
**Pull request by dbb-aws-only**
*Issue #, if available:*
https://app.asana.com/0/1202938635037719/1204347624456378/f

*Description of changes:*
Fix integration test for Glacier Table Creation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py [hunk 1]
```diff
@@ -23,14 +23,33 @@ def test_table_access_pattern_and_partition_key() -> None:
     table_name = os.environ[OutputKeys.GLACIER_RETRIEVAL_TABLE_NAME]
     client: DynamoDBClient = boto3.client("dynamodb")
 
-    key = {"pk": {"S": "pk-testing"}, "sk": {"S": "sk-testing"}}
-    value = str(uuid.uuid4())
+    items = []
+    for i in range(9):
+        value = str(uuid.uuid4())
+        key = {"pk": {"S": "pk-testing"}, "sk": {"S": f"sk:{i}"}}
```
Comment (bassemwanis) [inline source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:29]: You are not using `key` variable anywhere.

File: source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py [hunk 2]
```diff
@@ -23,14 +23,32 @@ def test_table_access_pattern_and_partition_key() -> None:
...
+    items = []
+    for i in range(9):
+        value = str(uuid.uuid4())
+        item = {
+            "pk": {"S": "pk-testing"},
+            "sk": {"S": f"sk:{i}"},
+            "testing_value": {"S": value},
+        }
+        client.put_item(TableName=table_name, Item=item)
+        items.append(item)
```
Comment (kamyarz-aws) [inline source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:35]: Is it possible for this the be reused in other tests? You could add this as a fixture to the conftest.

File: source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py [hunk 3]
```diff
@@ -23,14 +23,32 @@ def test_table_access_pattern_and_partition_key() -> None:
...
+    for item in items:
+        client.delete_item(TableName=table_name, Key=item["pk"])
+
+    response = client.query(
+        TableName=table_name,
+        KeyConditionExpression="pk = :pk",
+        ExpressionAttributeValues={":pk": {"S": "pk-testing"}},
     )
-    client.delete_item(TableName=table_name, Key=key)
+    assert len(response["Items"]) == 0
```
Comment (kamyarz-aws) [inline source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:54]: Do you want this to make sure that the table has no item or is it part of this test? if the former, this needs to be a part of a clean up function. if latter. This needs to be a different test.

Comment (dbb-aws-only) [reply source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:55]: At the end of the test, I delete only the items that I added and ensure that all of those items are gone.

Comment (kamyarz-aws) [reply source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:55]: So yeah just create a clean up function for it. its part of the integration and unit test modules/classes to have a set up function and/or clean up functions. You could have the clean up function here.

File: source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py (hunk 1 above)
Comment (dbb-aws-only) [reply source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:29]: I've decided to keep the key variable and use it in the Delete test.

File: source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py (hunk 3 above)
Comment (dbb-aws-only) [reply source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py]: The "Delete" step is an actual test (in this very specific case) - just to ensure that the dynamodb / iam policies are correct.  I'll create setup / cleanup sections for my more generic tests.

File: source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py [hunk 4]
```diff
@@ -22,14 +21,37 @@ def test_table_access_pattern_and_partition_key() -> None:
...
+        TableName=table_name,
+        KeyConditionExpression="pk = :pk",
+        ExpressionAttributeValues={":pk": {"S": "pk-testing"}},
     )
-    assert (
-        value
-        == client.get_item(TableName=table_name, Key=key)["Item"]["testing_value"]["S"]
+    assert len(response["Items"]) == 9
+    assert response["Items"][0] == items[0]
+    assert response["Items"][8] == items[8]
```
Comment (kamyarz-aws) [inline source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:47]: I dont understand the purpose of these two

Comment (dbb-aws-only) [reply source/tests/integration/infrastructure/test_glacier_object_retrieval_dynamodb.py:47]: It's sort of a random sample of the highest and lowest sorted items.  "Make sure the Lowest items match, make sure the Highest items match".  I thought checking the top and bottom would be sufficient.

**Pull request closed without merging**